lib/src/rules/missing_conditional_import.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Missing conditional import.';
+
+const _details = r'''
+
+**DON'T** reference files that do not exist in conditional imports.
+
+Code may fail at runtime if the condition evaluates such that the missing file
+needs to be imported.
+
+**BAD:**
+```dart
+import 'file_that_does_not_exist.dart';
+```
+
+**GOOD:**
+```dart
+import 'file_that_exists.dart';
+```
+
+''';
+
+class MissingConditionalImport extends LintRule {
+  MissingConditionalImport()
+      : super(
+            name: 'missing_conditional_import',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addCompilationUnit(this, visitor);
+  }
+}
+
+class _Visitor extends RecursiveAstVisitor<void> {
```

Comment:
For performance reasons, the linter uses visitors as a dispatch mechanism rather than as a tree walking mechanism. Walking the tree is expensive, so we do it once and effectively run all of the lint rules simultaneously. As a result,
- `_Visitor` should extend `SimpleAstVisitor`,
- you should use `addConfiguration` rather than `addCompilationUnit` on line 43, and
- you should remove the invocation of `super`

Reply:
Ah, apparently the file I duplicated as a starting point was the only one using a RecursiveAstVisitor (`missing_whitespace_between_adjacent_strings.dart`) 😄 

This makes sense and I've made those changes, thanks!

Reply:
That rule should definitely be updated. (But obviously not an issue for this PR.)

Reply:
Ah, wasn't sure if there was a reason. I'll file a PR, seems like a trivial change. Thanks!